flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/sink/KafkaRecordSerializationSchemaBuilderTest.java

```diff
@@ -256,6 +271,132 @@ public void testSerializeRecordWithTimestamp() {
         assertThat(recordWithInvalidTimestamp.timestamp()).isNull();
     }
 
+    @Test
+    public void testGetLineageDatasetFacetsWhenTopicSelectorNotKafkaTopicsIdentifierProvider() {
+        SerializationSchema<String> serializationSchema = new SimpleStringSchema();
+        KafkaRecordSerializationSchema<String> schema =
+                KafkaRecordSerializationSchema.builder()
+                        .setTopicSelector((TopicSelector<Object>) o -> DEFAULT_TOPIC)
+                        .setValueSerializationSchema(serializationSchema)
+                        .setKeySerializationSchema(serializationSchema)
+                        .build();
+
+        assertThat(((KafkaDatasetFacetProvider) schema).getKafkaDatasetFacet()).isEmpty();
```

Comment:
A bit more assertj-ish would be

```
        assertThat(schema)
                .asInstanceOf(InstanceOfAssertFactories.type(KafkaDatasetFacetProvider.class))
                .returns(List.of(), KafkaDatasetFacetProvider::getKafkaDatasetFacet);
```

That would result in an assertion error instead of runtime error if the Schema does not implement the interface.